src/main.c

```diff
@@ -186,6 +186,27 @@ PPPD_USAGE \
 "      trusted-cert = othercertificatedigest6631bf...\n" \
 "  For a full-featured configuration see man openfortivpn(1).\n"
 
+/**
+ * Returns the given "input" prefixed with "prefix" in a dynamically
+ * allocated string. This behaves exactly like "strdup" if "input" already
+ * starts with "prefix".
+ */
+static char *strdup_with_prefix(const char *input, const char *prefix)
+{
+	size_t prefix_len = strlen(prefix);
+	char *output;
+
+	if (memcmp(prefix, input, prefix_len) == 0)
```

Comment:
Technically this allows `memcpy` to read `prefix_len` bytes starting from `input`, which is wrong in case `input` is shorter than that. This is why I was comparing the lengths of the strings. Maybe you wanted to use `strncmp`? I used the `mem*` functions simply because I needed to compute the length of everything already.

Reply:
Right, `strcmp()` it is.